editor/src/components/canvas/controls/select-mode/controls-common.tsx

```diff
@@ -293,3 +298,36 @@ export function shouldShowControls(
 
   return true
 }
+
+export function useOnControlHover(controlId: string) {
+  const selectedElement = useEditorState(
+    Substores.metadata,
+    (store) => store.editor.selectedViews?.[0],
+    'selectedElement',
+  )
+  const setHoveredInspectorControl = useSetAtom(HoveredInspectorControlAtom)
+  const setGapHovered = React.useCallback(
+    (isHovered: boolean) => {
+      setHoveredInspectorControl(
+        isHovered && selectedElement != null
+          ? `${EP.toString(selectedElement)}-${controlId}`
```

Comment:
I think it'd be worth factoring out this string interpolation to a small helper function to be reused by e.g. line 328